src/lazy/text/buffer.rs

```diff
@@ -331,6 +411,74 @@ impl<'data> TextBufferView<'data> {
         .parse(self)
     }
 
+    /// Matches a list.
```

Comment:
🗺️ Introduces a matching function for the `list` type that finds its end offset.